src/io/flutter/logging/FlutterLogTree.java

```diff
@@ -482,6 +493,48 @@ public FlutterLogTree(@NotNull FlutterApp app,
     super(model);
     model.setTree(this.getTree());
     this.model = model;
+    registerShortcut();
+  }
+
+  public void sendToClipboardSelectedItems() {
+    ApplicationManager.getApplication().invokeLater(() -> {
+      final int[] rows = getSelectedRows();
+      final FlutterLogTree.LogRootTreeNode root = getLogTreeModel().getRoot();
+      final TreePath[] paths = getTree().getSelectionPaths();
+      if (paths != null) {
+        final Clipboard clipboard = Toolkit.getDefaultToolkit().getSystemClipboard();
+        final StringBuilder sb = new StringBuilder();
+        for (final TreePath path : paths) {
+          final Object pathComponent = path.getLastPathComponent();
+          if (pathComponent instanceof FlutterLogTree.FlutterEventNode) {
+            ((FlutterLogTree.FlutterEventNode)pathComponent).describeTo(sb);
+          }
+        }
+
+        final StringSelection selection = new StringSelection(sb.toString());
+        clipboard.setContents(selection, selection);
+      }
+    });
+  }
+
+  private void registerShortcut() {
+    final AnAction actionCopy = ActionManager.getInstance().getAction(IdeActions.ACTION_COPY);
+    final ShortcutSet copyShortcutSet = actionCopy.getShortcutSet();
```

Comment:
Query shortcut for copy action then register to FlutterLogTree.
When user trigger that action -> we simply call `sendToClipboardSelectedItems`

Reply:
I need to think about this some more but a quick question: how does this handler get **de**-registered?

Curious how this is done in other views... Or maybe you looked?

Reply:
@pq, `de-registered` mean?

> Curious how this is done in other views... Or maybe you looked?

Currently, the default action copy/paste it copies string values of the selected row in the tree view. But the string take by method
```java
com.intellij.ui.treeStructure.treetable.ListTreeTableModelOnColumns#getValueAt
  public Object getValueAt(Object value, int column) {
    return myColumns[column].valueOf(value);
  }
```

In our code
```java
      @Nullable
      @Override
      public FlutterLogEntry valueOf(DefaultMutableTreeNode node) {
        if (node instanceof FlutterEventNode) {
          return ((FlutterEventNode)node).entry;
        }
        return null;
      }
```

=> we return `FlutterLogEntry` for each cell.
=> when user copy => the result looks like  this:
```
io.flutter.logging.FlutterLogEntry@76b5b7da | io.flutter.logging.FlutterLogEntry@76b5b7da | io.flutter.logging.FlutterLogEntry@76b5b7da | io.flutter.logging.FlutterLogEntry@76b5b7da
```

Reply:
Another solution is
```
      @Nullable
      @Override
      public FlutterLogEntry valueOf(DefaultMutableTreeNode node) {
        if (node instanceof FlutterEventNode) {
          // TODO: Instead return FlutterLogEntry we will return STRING value for each column.
          return ((FlutterEventNode)node).entry;

        }
        return null;
      }
```